brian2/codegen/permutation_analysis.py

```diff
@@ -33,10 +34,11 @@ def check_for_order_independence(statements, variables, indices):
     # only from the values of the derived indices. In the most common case of Synapses, the main index would be
     # the synapse index, and the derived indices would be pre and postsynaptic indices (which can be repeated).
     main_index_variables = set([v for v in variables
-                                if (indices[v] in ('_idx', '0')
-                                    or getattr(variables[indices[v]],
-                                               'unique',
-                                               False))])
+                                if (not isinstance(variables[v], Constant) and
+                                    (indices[v] in ('_idx', '0')
+                                     or getattr(variables[indices[v]],
+                                                'unique',
+                                                False)))])
```

Comment:
Could you explain the logic of this change a bit to help me review it? Won't this change mean that constants will just be moved into ``different_index_variables`` which feels like the wrong behaviour?